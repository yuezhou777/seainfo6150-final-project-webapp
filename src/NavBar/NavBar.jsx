import React from 'react';
import { Link } from "react-router-dom";
import styles from './NavBar.module.css';

const NavBar = () => {

    return(
        <div className={styles.navContainer}>
            <header className={styles.navContainer}>
                <nav>
                    <Link to="/" className={styles.navIcon}>PurrPet</Link>
                </nav>

                <nav>
                    <ul className={styles.navList}>
                        <li className={styles.navItem}>
                            <Link to="/aboutus" className={styles.link}>About Us</Link>
                        </li>

                        <li className={styles.navItem}>
                            <Link to="/allpets" className={styles.link}>All Pets</Link>
                        </li>
                    
                        <li className={styles.navItem}>
                            <Link to="/category" className={styles.link}>Categories</Link>
                        </li>

                        <li className={styles.navItem}>
                            <Link to="/contactus" className={styles.link}>Contact Us</Link>
                        </li>
                    </ul>
                </nav>
            </header>
        </div>
    );

}

export default NavBar;