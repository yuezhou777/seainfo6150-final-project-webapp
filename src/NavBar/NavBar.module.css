.navContainer {
    width: 100%;
    display: flex;
    flex-flow: row;
    flex-wrap: wrap;
    justify-content: space-between;
    margin: 10px 40px 20px 20px;
}

.navIcon {
    font-family: 'Gill Sans', sans-serif;
    font-size: 3rem;
    font-weight: bold;
    color: orange;
}

.navList {
    display: flex;
    text-align: justify;
}

.navItem {
    font-family: 'Gill Sans', sans-serif;
    font-size: 1.5rem;
    display: block;
    color: white;
    padding: 14px 16px;
}

.link {
    color: olive;
    font-weight: bold;
}

.navItem:hover {
    background-color:greenyellow;
}

