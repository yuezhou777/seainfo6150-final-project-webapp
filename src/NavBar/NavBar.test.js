import React from 'react'
import {render} from "@testing-library/react";
import NavBar from "./NavBar.jsx";
import { BrowserRouter } from 'react-router-dom';

describe('test nav bar', () => {
    it('render NavBar successfully',  () => {
        const {container} = render(
            <BrowserRouter>
                <NavBar />
            </BrowserRouter>
        );

        expect(container).toMatchSnapshot();
    });
});