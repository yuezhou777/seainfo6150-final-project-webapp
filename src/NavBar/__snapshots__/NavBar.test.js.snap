// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test nav bar render NavBar successfully 1`] = `
<div>
  <div
    class="navContainer"
  >
    <header
      class="navContainer"
    >
      <nav>
        <a
          class="navIcon"
          href="/"
        >
          PurrPet
        </a>
      </nav>
      <nav>
        <ul
          class="navList"
        >
          <li
            class="navItem"
          >
            <a
              class="link"
              href="/aboutus"
            >
              About Us
            </a>
          </li>
          <li
            class="navItem"
          >
            <a
              class="link"
              href="/allpets"
            >
              All Pets
            </a>
          </li>
          <li
            class="navItem"
          >
            <a
              class="link"
              href="/category"
            >
              Categories
            </a>
          </li>
          <li
            class="navItem"
          >
            <a
              class="link"
              href="/contactus"
            >
              Contact Us
            </a>
          </li>
        </ul>
      </nav>
    </header>
  </div>
</div>
`;
